File: ExamenUnidad2_PYTHON 1/ExamenUnidad2_PYTHON 1/ExamenUnidad2/Empleados.py
from Persona import Persona

class Empleados(Persona):
    IDENTIFICADOR = 100

    def __init__(self, name, apellidos, fecha_de_nacimiento, CURP,
                 fecha_de_ingreso_como_trabajador, RFC, salary, horario, rol):
        super().__init__(name, apellidos, fecha_de_nacimiento, CURP)
        self.fecha_de_ingreso_como_trabajador = fecha_de_ingreso_como_trabajador
        self.RFC = RFC
        self.salary = salary
        self.horario = horario
        self.rol = rol
        self.id = Empleados.IDENTIFICADOR
        Empleados.IDENTIFICADOR += 1
        if rol=="Guia":
            self.esta_en_visita=False
    

    def get_fecha_de_ingreso_como_trabajador(self):
        return self.fechaDeIngresoComoTrabajador

    def set_Fecha_De_Ingreso_Como_Trabajador(self, fechaDeIngresoComoTrabajador):
        self.fechaDeIngresoComoTrabajador = fechaDeIngresoComoTrabajador

    def get_rfc(self):
        return self.RFC

    def set_rfc(self, RFC):
        self.RFC = RFC

    def get_salary(self):
        return self.salary

    def set_salary(self, salary):
        self.salary = salary

    def get_horario(self):
        return self.horario

    def set_horario(self, horario):
        self.horario = horario

    def get_rol(self):
        return self.rol

    def set_rol(self, rol):
        self.rol = rol

    def get_numero_de_rol(self):
        return self.numeroDeRol

    def set_numero_de_rol(self, numeroDeRol):
        self.numeroDeRol = numeroDeRol

    @staticmethod
    def get_identificador():
        return Empleados.IDENTIFICADOR

    @staticmethod
    def set_identificador(IDENTIFICADOR):
        Empleados.IDENTIFICADOR = IDENTIFICADOR

    def get_id(self):
        return self.id

    def set_id(self, id):
        self.id = id

    def obtener_informacion_empleado(self):
        return (f"\nId: {self.id} \nNombre: {self.get_name()} \nApellido: {self.get_apellidos()} \n"
        f"Fecha de nacimiento: {self.get_fecha_de_nacimiento()} \nCurp: {self.get_CURP()} \n"
        f"Fecha de ingreso como trabajador: {self.fecha_de_ingreso_como_trabajador}")
    
    def get_esta_en_visita(self):
        return self.get_esta_en_visita

    def set_esta_en_visita(self, esta_en_visita):
        self.esta_en_visita = esta_en_visita

File: ExamenUnidad2__JAVA/ExamenUnidad2/src/examenunidad2/Empleados.java
package examenunidad2;

public class Empleados extends Persona
{
    public static int IDENTIFICADOR = 100;
    private String fechaDeIngresoComoTrabajador;
    private String RFC;
    private double salary;
    private String horario;
    private String rol;
    private int numeroDeRol ;//
    private int id;
    private boolean estaEnVisita;
    public Empleados(String name, String apellidos, String fechaDeNacimiento, String CURP,
            String fechaDeIngresoComoTrabajador, String RFC, double salary, String horario, String rol) 
    {
        super(name, apellidos, fechaDeNacimiento, CURP);
        this.fechaDeIngresoComoTrabajador = fechaDeIngresoComoTrabajador;
        this.RFC = RFC;
        this.salary = salary;
        this.horario = horario;
        this.rol = rol;
        this.id = IDENTIFICADOR +=1;
        if(rol.equals("Guía"))
            this.estaEnVisita=false;
    }
    
    public Empleados(String name, String apellidos, String fechaDeNacimiento, String CURP, int numeroDeRol) 
    {
        super(name, apellidos, fechaDeNacimiento, CURP);
        this.numeroDeRol = numeroDeRol;
    }

    public String getFechaDeIngresoComoTrabajador() {
        return fechaDeIngresoComoTrabajador;
    }

    public void setFechaDeIngresoComoTrabajador(String fechaDeIngresoComoTrabajador) {
        this.fechaDeIngresoComoTrabajador = fechaDeIngresoComoTrabajador;
    }

    public String getRFC() {
        return RFC;
    }

    public void setRFC(String RFC) {
        this.RFC = RFC;
    }

    public double getSalary() {
        return salary;
    }

    public void setSalary(double salary) {
        this.salary = salary;
    }

    public String getHorario() {
        return horario;
    }

    public void setHorario(String horario) {
        this.horario = horario;
    }

    public String getRol() {
        return rol;
    }

    public void setRol(String rol) {
        this.rol = rol;
    }

    public int getNumeroDeRol() {
        return numeroDeRol;
    }

    public void setNumeroDeRol(int numeroDeRol) {
        this.numeroDeRol = numeroDeRol;
    }

    public static int getIDENTIFICADOR() {
        return IDENTIFICADOR;
    }

    public static void setIDENTIFICADOR(int IDENTIFICADOR) {
        Empleados.IDENTIFICADOR = IDENTIFICADOR;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
    
    
    public String obtenerInformacionEmpleado()
    {
        return String.format("\nId: %d \nNombre: %s \nApellido: %s \nFecha de nacimiento: %s \nCurp: %s \nFecha de ingreso como trabajdor: %s \nRFC: %s \nSalario: %.4f \nHorario: %s \nRol: %s ",id, getName(), getApellidos(), getFechaDeNacimiento(), getCURP(),
             fechaDeIngresoComoTrabajador, RFC,  salary, horario, rol);
    }

    public boolean getEstaEnVisita() {
        return estaEnVisita;
    }

    public void setEstaEnVisita(boolean estaEnVisita) {
        this.estaEnVisita = estaEnVisita;
    }
    
    
}
